src/main/scala/viper/gobra/ast/internal/Program.scala

```diff
@@ -234,7 +234,7 @@ sealed trait MakeStmt extends Stmt {
 }
 
 case class MakeSlice(override val target: LocalVar, override val typeParam: SliceT, lenArg: Expr, capArg: Option[Expr])(val info: Source.Parser.Info) extends MakeStmt
-case class MakeChannel(override val target: LocalVar, override val typeParam: ChannelT, bufferSizeArg: Option[Expr], isChannel: MPredicateProxy)(val info: Source.Parser.Info) extends MakeStmt
+case class MakeChannel(override val target: LocalVar, override val typeParam: ChannelT, bufferSizeArg: Option[Expr], isChannel: MPredicateProxy, bufferSize: MethodProxy)(val info: Source.Parser.Info) extends MakeStmt
```

Comment:
Why do we need to pass the proxy for `BufferSize` here? Won't this field be the same to every instance of `MakeChannel`?

Reply:
It is the same for all `MakeChannel` having the same channel element type. The reason to attach the proxy to the internal node comes from our current architecture for built-in members: the desugarer notices that the buffer size method is used, creates a unique name for it, attaches the corresponding proxy to the `MakeChannel` node, and keeps track of all built-in members used so far. This will then trigger the encoding to create a corresponding encoding for the buffer size method. The advantage of this architecture is that the built-in member's call site (here the `make` call) does not need to differentiate between calling a built-in and a non-built-in member